index.js

```diff
@@ -65,15 +65,16 @@ var Block = module.exports = function (data, opts) {
     rawUncleHeaders = data.uncleHeaders || []
   }
 
+  const height = new BN(this.header.number).toNumber()
+  const hardfork = this._common.activeHardfork(height)
+  this._common.setHardfork(hardfork)
```

Comment:
Hmm, this can't be the solution yet, now we have got inconsistent hardfork setting logic in block and header which can eventually end up in differing common objects.

Not sure, maybe we need to pass options to `header` explicitly and use the eventually altered `_common` object directly or something. Or a unified `util` function used both by `block` and `header` - eventually from a separate `util.js` file or something getting for the logic of creating or taking the common object, so that this is unified?

Just some ideas, but we have to make sure that on 100% both end up with the same `Common` state.

Reply:
I see what you mean. I'll mess around with it and see if I can get them to be synchronized

Reply:
I tried making the BlockHeader's common object set the fork, then making block just inherit the common object from the header, but 15k tests fail if you do that. I don't really know enough to understand why.